src/librustc_metadata/decoder.rs

```diff
@@ -1542,13 +1541,9 @@ pub fn is_extern_item<'a, 'tcx>(cdata: Cmd,
     let applicable = match item_family(item_doc) {
         ImmStatic | MutStatic => true,
         Fn => {
-            let ty::TypeScheme { generics, ty } = get_type(cdata, id, tcx);
+            let ty::TypeScheme { generics, ty: _ } = get_type(cdata, id, tcx);
```

Comment:
You should be able to actually change this to:

```rust
let ty::TypeScheme { generics, _ } = ...
```

that is, remove the `ty` matching entirely